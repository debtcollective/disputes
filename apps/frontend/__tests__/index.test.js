/* eslint-env jest */

import App from "../pages/index.js";
import React from "react";
import { render } from "react-testing-library";
import wait from "waait";

describe("With React Testing Library", () => {
  it("Shows \"Hello world!\"", () => {
    const { getByText } = render(<App />);

    expect(getByText("General Dispute")).not.toBeNull();
  });
});

describe("With React Testing Library Snapshot", () => {
  it("Should match Snapshot", async () => {
    const { asFragment } = render(<App />);

    // wait for next tick to test final state
    // https://www.apollographql.com/docs/react/recipes/testing.html#Testing-final-state
    await wait(0);

    expect(asFragment()).toMatchSnapshot();
  });
});
