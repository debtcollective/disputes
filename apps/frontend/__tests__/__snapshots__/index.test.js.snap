// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      class="pages__HelloWorld-sc-6ucdmv-0 kFhcAB"
    >
      Hello World!
    </p>
    <ul>
      <li>
        <a
          href="posturl1"
        >
          1. title 1
        </a>
      </li>
      <li>
        <a
          href="posturl2"
        >
          2. title 2
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
